src/yunohost/backup.py

```diff
@@ -2371,6 +2371,12 @@ def backup_info(name, with_details=False, human_readable=False):
         if "size_details" in info.keys():
             for category in ["apps", "system"]:
                 for name, key_info in info[category].items():
+                    # Retrocompatibility with archives produced 
+                    # around january 2019
+                    if not isinstance(key_info, dict):
+                        key_info = {x: 'succeed' for x in key_info}
+                        info[category][name] = key_info
+                        
                     if name in info["size_details"][category].keys():
```

Comment:
Hmyea in fact I don't know if it's so interesting to have all these `{'path1':'succeed', 'path2':'succeed'}` ... I'm not even sure that the value can be different than suceed ... So maybe having the dict is a buggy behavior (probably my fault ?) that wasn't noticed when implementing the whole `size_details` stuff and we should just stick with a list ?

Reply:
Indeed, only succeed are listed in info :
https://github.com/YunoHost/yunohost/blob/stretch-unstable/src/yunohost/backup.py#L613

Reply:
I try to create an history of this format:
2.4:
```
{"hooks": {"conf_ldap": ["/usr/share/yunohost/hooks/backup/05-conf_ldap"]}, "created_at": 1494198169, "apps": {}, "description": "", "size": 524892}
```

3.5, size_details and hooks is replaced by system
```
{"size_details": {"apps": {}, "system": {"conf_ldap": 3692}}, "description": "", "created_at": 1546422083, "apps": {}, "system": {"conf_ldap": ["/usr/share/yunohost/hooks/backup/05-conf_ldap"]}, "size": 116924}
```

Current 3.6, use a dict instead of a list
```
{"size_details": {"apps": {}, "system": {"conf_ldap": 4134}}, "description": "", "created_at": 1562680193, "apps": {}, "system": {"conf_ldap": {"/usr/share/yunohost/hooks/backup/05-conf_ldap": "succeed"}}, "size": 118861}
```

So now, if we change again, it will complexify the code. We could also decide to stop retrocompatibility of 2.4 format to simplify some code or decide to hotfix quickly 3.6 and ignore this new 3.6 format.

In a perfect world, and with the current features (not the future one's), i think the format should be:
```
{"description": "", "created_at": 1562680193, "apps": {}, "system": {"conf_ldap": {"hook": "/usr/share/yunohost/hooks/backup/05-conf_ldap", "size": 4134}}, "size": 118861}
```
So no size_details, and a dict to put some meta info about this "part of backup".

Reply:
>In a perfect world, and with the current features (not the future one's), i think the format should be:
>
>`"system": {"conf_ldap": {"hook": "/usr/share/yunohost/hooks/backup/05-conf_ldap", "size": 4134}}`

We can't have this, because the value associated to `conf_ldap` could contain several paths ... so at least the value corresponding to `hook` should be a list (meh). That's why in 3.6 it's a dict of path ... but imho having a dict with just "succeed" as value is a bug of 3.6, and we should just revert to having simple list of paths ...

Reply:
Note, it could be a good idea to register in info, system_part and apps failed to backup... Indeed in webadmin it could be usefull to display a backup and see part in failures... So the idea of succeed/failed is not so bad.

And after some test, i discover it's possible to have several hooks attached to the same system part ! That's why a list was used and not just a string...

Reply:
> so at least the value corresponding to hook should be a list (meh)
It's exactly what the PR does

Reply:
(Follow-up : we discussed this IRL and this is not what the PR does. Ideally we should revert the thing to not store those useless "success" flags in the info.json, while still adding legacy code to support the buggy/legacy 3.5~3.6 format)